backend/core/socketio_config.py

```diff
@@ -202,7 +204,84 @@ def handle_incoming_message(session: Session, message: dict):
+
+
+def update_chat_status(session: Session, message: dict):
+    conversation_envelope = get_value_or_raise_error(
+        message, "conversation_envelope"
+    )
+    chat_id = conversation_envelope.get("message_id", None)
+    logger.info(f"Updating chat {chat_id} status")
+    chat = session.exec(select(Chat).where(Chat.id == chat_id)).first()
+    if not chat:
+        return None
+    chat.status = 1
+    session.commit()
+    session.flush()
+    logger.info(f"Chat {chat_id} status updated")
+
+
+async def resend_messages(session: Session, user_id=int, user_sid=str):
+    chat_session = session.exec(
+        select(Chat_Session).where(Chat_Session.user_id == user_id)
+    ).all()
+    if not chat_session:
+        return None
+    last_ten_chats = session.exec(
+        select(Chat)
+        .where(
+            and_(
+                Chat.chat_session_id.in_([cs.id for cs in chat_session]),
+                Chat.sender_role.in_(
+                    [Sender_Role_Enum.CLIENT, Sender_Role_Enum.ASSISTANT]
+                ),
+            )
+        )
+        .order_by(Chat.created_at.desc())
+        .limit(10)
+    ).all()
+    # Reorder the results by created_at in ascending order
+    last_ten_chats = sorted(last_ten_chats, key=lambda x: x.created_at)
+    for chat in last_ten_chats:
+        # starting to resend message
+        media = []
+        context = []
+        if chat.media:
+            for cm in chat.media:
+                media.append({"url": cm.url, "type": cm.type})
+                context.append(
+                    {
+                        "url": cm.url,
+                        "type": cm.type,
+                        "caption": chat.message,
+                    }
+                )
+        message = queue_message_util.create_queue_message(
+            chat_session_id=chat.chat_session_id,
+            message_id=chat.id,
+            client_phone_number=f"+{chat.chat_session.client.phone_number}",
+            user_phone_number=f"+{chat.chat_session.client.phone_number}",
+            sender_role=chat.sender_role,
+            sender_role_enum=Sender_Role_Enum,
+            platform=chat.chat_session.platform,
+            platform_enum=Platform_Enum,
+            body=chat.message,
+            media=media,
+            context=context,
+            timestamp=chat.created_at.isoformat() + "+00:00",
+        )
+        if chat.sender_role == Sender_Role_Enum.CLIENT:
+            await sio_server.emit(
+                "chats", message, to=user_sid, callback=emit_chats_callback
+            )
+            logger.info(f"Resend message for client->user: {message}")
+        if chat.sender_role == Sender_Role_Enum.ASSISTANT:
+            await sio_server.emit(
+                "whisper", message, to=user_sid, callback=emit_whisper_callback
+            )
+            logger.info(f"Resend message for assistant->user: {message}")
+    return last_ten_chats
 
 
 async def user_to_client(body: str):
```

Comment:
This method does not store the user message in the database, which I think it should, right?

Reply:
we store all off user message to database here: https://github.com/akvo/rag-doll/blob/e0447f25bb4a2c5eb437f79df3901959f3451061/backend/core/socketio_config.py#L276-L282

Reply:
Would it note make more sense to have it central, now you have to save Slack and WhatsApp messages separately. Once you get to SMS, that will make a 3rd place. If you save here, there is only 1 place instead of 2.